File: easy/first-bad-version/first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        startVersion = 1
        endVersion = n
        while(startVersion <= endVersion):
            mid = (startVersion + endVersion) // 2
            if isBadVersion(mid):
                endVersion = mid - 1
            else:
                startVersion = mid + 1
        return startVersion

File: easy/first-bad-version/first-bad-version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n)
{
    unsigned int start_version = 1;
    unsigned int end_version = n;
    while (start_version <= end_version)
    {
        int mid = (start_version + end_version) >> 1;
        if (isBadVersion(mid))
        {
            end_version = mid - 1;
        }
        else
        {
            start_version = mid + 1;
        }
    }
    return start_version;
}